src/pages/edge/adobe-journey-optimizer/code-based/tutorial.md

```diff
@@ -0,0 +1,114 @@
+---
+title: Automatically display and track push notification
+description: This document describes how to automatically display and track push notifications using the AEPMessaging extension. You should to use this method unless you have a specific requirement to manually display and track push notifications.
+keywords:
+- Adobe Journey Optimizer
+- Messaging
+- Code-based experiences
+- Tutorial
+---
+
+import Tabs from './tabs/tutorial.md'
+
+# Code-based experiences implementation tutorial
+
+This document describes how to fetch, display and track code-based experiences  using the Adobe Journey Optimizer extension.
+
+## Pre-requisites
+
+[Integrate and register Messaging extension](../../../index.md#implement-extension-in-mobile-app) in your app.
+
+## Fetch and cache the code-based content
+
+To fetch the content for the surfaces configured in Adobe Journey Optimizer campaigns, call the [updatePropositionsForSurfaces](../api-reference.md#updatepropositionsforsurfaces) API . It is encouraged to batch requesting multiple [Surface](../public-classes/surface.md) URIs in a single API call when possible. The returned code-based experiences are cached in-memory by the Messaging extension and persists through the lifecycle of the app (i.e as long as the app is running). An example of the call is shown below.
```

Comment:
I personally don't love the phrasing of "It is encouraged..."

I'd rather go with "You should batch request multiple..."

Reply:
"An example of the call is shown below:"